File: Shell Sort/shell_sort.py
def ShellSort(arr):
    h = len(arr) // 2
    while h > 0:
        for i in range(h, len(arr)):
            t = arr[i]
            j = i
            while j >= h and arr[j - h] > t:
                arr[j] = arr[j - h]
                j -= h
            arr[j] = t
        h = h // 2
    return arr

File: Shell Sort/shell_sort.js
const ShellSort = (arr) => {
  for (let i = arr.length / 2; i > 0; i = Math.floor(i / 2))
    for (let j = i; j < arr.length; ++j)
      for (let k = j - i; k >= 0 && arr[k] > arr[k + i]; k -= i)
        [arr[k], arr[k + i]] = [arr[k + i], arr[k]];
  return arr;
};
